src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1259,18 +1238,12 @@ public Block FindBlock(Keccak blockHash, BlockTreeLookupOptions options)
                 return null;
             }
 
-            Block block = _blockCache.Get(blockHash);
+            Block block = _blockDb.Get(blockHash, _blockDecoder, _blockCache, false);
```

Comment:
just important thing to remember is not to cache headers that do not have total diffuclty set

Reply:
yep there is explicit false not to cache at this call